hsds/servicenode_lib.py

```diff
@@ -1013,41 +1017,78 @@ async def deleteObj(app, obj_id, bucket=None):
         del meta_cache[obj_id]  # remove from cache
 
 
-async def createGroup(app, root_id=None, creation_props=None, bucket=None):
-    """ create a group object and return group json """
-
-    group_id = createObjId("groups", rootid=root_id)
-    log.info(f"new group id: {group_id}")
-    group_json = {"id": group_id, "root": root_id}
+async def createObject(app,
+                       root_id=None,
+                       obj_type=None,
+                       obj_shape=None,
+                       layout=None,
+                       creation_props=None,
+                       bucket=None):
+    """ create a group, ctype, or dataset object and return group json
+        Determination on whether a group, ctype, or dataset is created is based on:
+            1) if obj_type and obj_shape are set, a dataset object will be created
+            2) if obj_type is set but not obj_shape, a  datatype object will be created
+            3) otherwise (type and shape are both None), a group object will be created
+        The layout parameter only applies to dataset creation
+    """
+    if obj_type and obj_shape:
```

Comment:
Is there reason to determine what kind of object to create in this way, instead of just using `obj_type` to carry that information?

Reply:
Obj_type is actually the type json fir a dataset or committed data type not an identifier to distinguish between group dataset and ctype   Guess the name is a bit confusing.